docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,83 @@
+- A tenancy-based read-only access to resources. 
+- Users should be allowed to grant editing capabilities to their resources to other users in their organization.
+
+## Goals
+The following tables details the access permission rules for various resources:
+
+For clusters and cluster-based resources (all APIs that starts with `/v2/clusters/{cluster_id}`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List Cluster   | Returns clusters owned by all members of the organization                                      |
+| Get  Cluster   | Allowed for all members of the organization                                                    |
+| Edit Cluster   | Allowed for the owner and for each user that has update access for this cluster's subscription |
+| Delete Cluster | Allowed for the owner and for each user that has remove access for this cluster's subscription |
+
+For InfraEnv and InfraEnv-based resources like hosts (all APIs that starts with `/v2/infra-envs/{infra_env_id`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List InfraEnvs | For bound InfraEnvs, returns resources owned by all members of the organization<br>If unbound InfraEnvs, return resources if owned by the caller                                                                |
+| Get    InfraEnv| For bound InfraEnv, allowed for all members of the organization<br>For unbound InfraEnv allowed for the owner                                                                                                         |
+| Edit InfraEnv  | For bound InfraEnv, allowed for the owner and for each user that has update access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+| Delete InfraEnv| For bound InfraEnv, allowed for the owner and for each user that has remove access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+
+For day2 cluster operations:
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| Add Host       | Allowed for every user who has update access for the day1's cluster subscription               |
+
+Filtering for you own clusters:
+Get Cluster call shall have an additional query parameter which filters the cluster by owners only:
+`GET /v2/clusters?owneronly=true`
+
+## Authorization Flow
```

Comment:
This is the current one? If it is,it's not clear from the title

Reply:
Sarah, most readers aren't familiar with AMS and our integration with it.  I suggest a short background followed by another subsection on the changes.

Reply:
More details added.